metron-analytics/metron-maas-common/src/main/java/org/apache/metron/maas/discovery/ServiceDiscoverer.java

```diff
@@ -93,6 +116,7 @@ private void updateState() {
         }
       }
       rwLock.writeLock().lock();
+      this.modelToCurrentVersion = modelToVersion;
       this.state = state;
       this.containerToEndpoint = containerToEndpoint;
       rwLock.writeLock().unlock();
```

Comment:
There are a few other places where locks are used without try/finally.  For example in blacklist(...) and getEndpoint(...). May need to hit those up too.